src/ui/buttons/buttonDesigns/divideLogoAnimation.jsx

```diff
@@ -0,0 +1,154 @@
+/* @flow */
+/** @jsx node */
+import { LOGO_CLASS } from '@paypal/sdk-logos/src';
+import { node, Fragment, type ChildType } from 'jsx-pragmatic/src';
+
+import { CLASS } from '../../../constants';
+import { BUTTON_SIZE_STYLE } from '../config';
+
+import { type ContentOptions } from './types';
+
+type DivideLogoAnimationProps = {|
+    paypalLabelContainerElement : Object,
+    paypalLogoStartingLeftPosition : string
+|};
+
+
+export const DIVIDE_LOGO_CONFIG = {
+    min:                            BUTTON_SIZE_STYLE.large.minWidth,
+    max:                            BUTTON_SIZE_STYLE.huge.maxWidth,
+    cssUtilClasses: {
+        DOM_READY:                  CLASS.DOM_READY,
+        PAYPAL_LOGO:                LOGO_CLASS.LOGO,
+        PAYPAL_BUTTON_LABEL:        CLASS.BUTTON_LABEL
+    }
+};
+
+// Returns props necessary to render the animation as long as they are valid
+export const getDivideLogoProps = function (document : Object, configuration : Object) : DivideLogoAnimationProps | null {
+    const { PAYPAL_BUTTON_LABEL, PAYPAL_LOGO } = configuration.cssUtilClasses;
+
+    const designContainer = (document && document.querySelector('.personalized-design-container')) || null;
+    if (!designContainer) {
+        return null;
+    }
+
+    // return null if animation should not be played for the button size
+    const designContainerWidth = designContainer.offsetWidth;
+    if (designContainerWidth < configuration.min || designContainerWidth > configuration.max) {
+        return null;
+    }
+
+    // get the label container that animation will be applied to
+    const paypalLabelContainerElement = designContainer.querySelector(`.${ PAYPAL_BUTTON_LABEL }`) || null;
+    if (!paypalLabelContainerElement) {
+        return null;
+    }
+
+    // get starting position for element so it doesn't flicker when animation begins
+    const paypalLogoElement = (paypalLabelContainerElement && paypalLabelContainerElement.querySelector(`.${ PAYPAL_LOGO }`)) || null;
+    const paypalLogoStartingLeftPosition = paypalLogoElement
+        ? `${ (paypalLogoElement.offsetLeft / paypalLabelContainerElement.offsetWidth) * 100 }%`
+        : '44.5%';
```

Comment:
Nit: where does this number come from? a comment can help

Reply:
Eyeballed. Will replace with a check for existence and remove ternary